particle-iot/device-os
# Fix/hal/gen3 pinmode

### Problem

1. User LED is configured to OUTPUT mode when initializing.
2. Pin mode is changed to default mode after calling `analogWrite()` and `digitalWrite()`

Reference issue:
https://github.com/particle-iot/device-os/issues/1772

### Steps to Test
1. Flash example
2. If D7 is OUTPUT mode, the device will enter PANIC mode
3. Use a multimeter to measure OUTPUT value from D4 pin, it should be high now

### Example App

```c
void setup(void) {
    SPARK_ASSERT(getPinMode(D7) == PIN_MODE_NONE);

    pinMode(D4, OUTPUT);
    analogWrite(D4, 128);
    digitalWrite(D4, 1);
}
```

### References

[CH32946]

---

### Completeness
